### ISSUE TITLE
Preview v2 inconsistent behaviour around block-referenced list items

### ISSUE THREAD

**Issue by djradon** [dendronhq/dendron#1235]
## Please select the area your Bug applys to. (Multiple selections are Possible)

- [ ] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [x] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Describe the bug
Sometimes unordered list item references display text from above the list.

## To Reproduce

Paste the following text into a new note:

```
## intersections

notes about the intersection of two concepts

- feels like a separate note is appropriate, e.g. metaverse + blockchain
  - a macro could be appropriate too, that auto-includes headers and blocks that are tagged appropriately! ^macro
  - can be transcluded (i.e. "referenced" in Dendron-speak) into each of the component notes ^DHWVfFIaPjYv
  - but
    - do intersection notes go in an "intersections" top-level hierarchy ?  ^OGXtuAHTSmxf
      - or, in both places?
      - probably doesn't matter where, as long as they're referenced

![[#^macro]]

![[#^DHWVfFIaPjYv]]


```

When I preview the above I get

```
notes about the intersection of two concepts
```

I can construct lists that behave as expected, but if I keep adding more items and references, at some point things stop working correctly. Sorry I haven't figured out exactly what pushes preview over the edge!

## Expected behavior

Should see the list items referenced instead of the pre-list text

## Screenshots

![image](https://user-images.githubusercontent.com/5224156/131289562-92396796-429a-48e0-a4a7-731deed1eee6.png)

## Desktop (please complete the following information)
- Windows, VSCode 0.56.0

## Additional context

There's an additional issue where if preview is already open, sometimes it reports "Start anchor ^DHWVfFIaPjYv not found", but closing and re-previewing seems to fix that issue.

**Comment by hikchoi** [comment]
Thank you for the report. I can reproduce this.

We'll track this and let you know once it gets fixed.

As for _Start anchor not found_, this has to do with how we are indexing newly created block references. For now, if you run `Reload Webview`, it will pick up the changes and display them correctly. This is less than ideal, so we will be trying to find a solution to fix this. 🙏🏼

**Comment by hikchoi** [comment]
pinging @SeriousBug for block reference issue.

**Comment by SeriousBug** [comment]
Thanks for the report, I'll look into this.

**Comment by SeriousBug** [comment]
This issue has now been fixed in version 0.58. Please let us know if it still happens, otherwise this issue will close in a few days.
